spec_2/index.js

```diff
@@ -0,0 +1,75 @@
+const letters = ['_','A','B','C','D','E','F','G','H','I','J','K','L','M','N','O',
+'P','Q','R','S','T','U','V','W','X','Y','Z']
+
+const letterValues = [
+  ['_', 2, 0],
+  ['A', 9, 0],
+  ['B', 2, 0],
+  ['C', 2, 0 ],
+  ['D', 4, 0],
+  ['E', 12, 0],
+  ['F',2, 0],
+  ['G',3, 0],
+  ['H',2, 0],
+  ['I',9, 0],
+  ['J',1, 0],
+  ['K',1, 0],
+  ['L',4, 0],
+  ['M',2, 0],
+  ['N',6, 0],
+  ['O',8, 0],
+  ['P',2, 0],
+  ['Q', 1, 0],
+  ['R', 6, 0],
+  ['S',4, 0],
+  ['T',6, 0],
+  ['U',4, 0],
+  ['V', 2, 0],
+  ['W',2, 0],
+  ['X',1, 0],
+  ['Y',2, 0],
+  ['Z',1, 0]
+]
+
+
+  const frequencies = [[],[],[],[],[],[],[],[],[],[],[],[],[]]
+
+  const setFrequencies = () => {
+    letters.forEach((letter, i , a) => {
+      const newFrequency = letterValues[i][1]
+      let targetArray = frequencies[newFrequency]
+      targetArray.push(letter)
+    })
+  }
+
+  const calculateNewFrequency = character => {
```

Comment:
This function, at first glance, appears to be doing a whole lot of work.  The formatting makes it difficult to immediately understand the intent of the code.  The reliance on order for meaning in the letters array adds to the confusion - there's a lot of index access that I can't understand without constantly referring to the declaration...  If I were to pseudo code this, it would look something like:
```js
  // Obtain a reference to the tile
  // Find the current index of this character in the frequencies array, starting with the character's default frequency, and working backwards
  // Swap the tile into index - 1 in the frequency array
```
A solution, given my use of objects above, might look something like this:
```js
const updateFrequency = character => {
  const tile = letterValues.find( letter => letter.face === character )

  // I don't like this loop, it feels clunky, but aiming for a solution first...
  let searchIndex = tile.frequency
  while( searchIndex >= 0 ) {
    if( frequencies[ searchIndex ].includes( character ) ) {
      break;
    } 
    searchIndex--
  }

  frequencies[ searchIndex ] = frequencies[ searchIndex ].filter( tile => tile !== character )
  if( searchIndex > 0 ) {
    frequencies[ searchIndex - 1 ].push( character )
  }
}
```
With this implementation that clearly implements the individual logical steps, I would then refactor - there's potential to pull some of these chunks of logic out into their own functions, and opportunities for more concise, functional implementation of some of the iteration logic.